ReferenceTests/src/tests/examples2d.jl

```diff
@@ -1151,10 +1151,10 @@ end
 
     fig = Figure(size = (600, 600))
     # Create a recipe plot
-    ax, plot_top = heatmap(fig[1, 1], randn(10, 10))
+    ax, plot_top = heatmap(fig[1, 1], randn(10, 10), colormap = [:transparent])
     # Plot some recipes at the level below the contour
-    scatterlineplot_1 = scatterlines!(plot_top, 1:10, 1:10; linewidth = 20, markersize = 20, color = :red)
-    scatterlineplot_2 = scatterlines!(plot_top, 1:10, 1:10; linewidth = 20, markersize = 30, color = :blue)
+    scatterlineplot_1 = scatterlines!(ax, 1:10, 1:10; linewidth = 20, markersize = 20, color = :red)
+    scatterlineplot_2 = scatterlines!(ax, 1:10, 1:10; linewidth = 20, markersize = 30, color = :blue)
```

Comment:
This test failed because of what I'd say is undefined behavior - plotting to a primitive plot. 
Specifically because `data_limits` has a specific overload for Heatmap, thus using heatmap limits, while `boundingbox` attempts to recognize primitives by their lack of child plots. Since heatmap has children here, boundingbox uses scatter + lines for limits. 
Plotting to a primitive also results in other weirdness - the heatmap doesn't show and the axis uses tight limits.

Reply:
Oh, plotting to a primitive outside a recipe?!  That's something we shouldn't do in any case ;) Didn't realize we actually do this in the tests!